core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -239,4 +239,25 @@ message RDDStorageInfo {
 
 message RDDStorageInfoWrapper {
   RDDStorageInfo info = 1;
-}
\ No newline at end of file
+}
+
+message SQLPlanMetric {
```

Comment:
Should we define it in the `core` module? Maybe it's better to put it in the `sql` module?

Reply:
It is ok to define it in `core` module. There is "SQL" concept in core module anyway. 
Can we skip shading protobuf in SQL module in this way?

Reply:
If we define all proto messages in the core module,  I think we can skip shading protobuf in SQL module,  the generated message code can find the corresponding `shaded+relocated` dependencies in the core module